### PR TITLE
Improve NFT loading

### PR DESCRIPTION
**Pull request by nop33**
This PR addresses these issues:
- https://github.com/alephium/alephium-frontend/issues/658
- https://github.com/alephium/alephium-frontend/issues/28
- https://github.com/alephium/alephium-frontend/issues/425

### CODE REVIEW

File: apps/mobile-wallet/src/components/NFTThumbnail.tsx [hunk 1]
```diff
@@ -92,6 +134,37 @@ const NFTThumbnail = ({ nft, size }: NFTThumbnailProps) => {
...
+)
+
+const WebViewImage = ({ nft, size }: NFTThumbnailProps) => (
+  <WebView
+    source={{ html: `<img src="${nft.image}" />` }}
+    style={{ width: size, height: size, borderRadius: BORDER_RADIUS_SMALL, backgroundColor: 'transparent' }}
+    injectedJavaScript={
+      "const meta = document.createElement('meta'); meta.setAttribute('content', 'width=width, initial-scale=1, maximum-scale=1, user-scalable=1'); meta.setAttribute('name', 'viewport'); document.getElementsByTagName('head')[0].appendChild(meta); "
+    }
+    javaScriptEnabled={false}
```
Comment (mvaivre) [inline apps/mobile-wallet/src/components/NFTThumbnail.tsx:150]: So this should avoid any injection attack, correct?
And can you confirm that all that can be rendered in this webview is an  tag?

File: apps/mobile-wallet/src/components/NFTThumbnail.tsx [hunk 2]
```diff
@@ -82,6 +118,12 @@ const NFTThumbnail = ({ nft, size }: NFTThumbnailProps) => {
                   </AttributesGrid>
                 )}
               </ScreenSection>
+
+              {!isDataUri && (
+                <ScreenSection>
+                  <Button title={t('View full size')} onPress={() => openBrowserAsync(nft.image)} />
```
Comment (mvaivre) [inline apps/mobile-wallet/src/components/NFTThumbnail.tsx:124]: Q: what will happen if the `image` filed is a link to a malicious / fishing website? I have the same problem on the explorer. We can't do much about it, can we?

**Comment by mvaivre** [comment]
Two things I noticed:

1) glitches when loading the NFT list, as if some images are available in cache, but reloaded anyway.

https://github.com/alephium/alephium-frontend/assets/6715943/cf38f300-09f5-4412-8dd0-ed2ea7275f5b

2) Nit: if I keep the list open, the thumbnails re-render every minute or so.

File: apps/mobile-wallet/src/components/NFTThumbnail.tsx (hunk 1 above)
Comment (nop33) [reply apps/mobile-wallet/src/components/NFTThumbnail.tsx:150]: The only user content that we are rendering is the `nft.image`, which I am sanitizing. So, I am fairly confident that we are safe. I don't see how anything could be injected in the webview

File: apps/mobile-wallet/src/components/NFTThumbnail.tsx (hunk 2 above)
Comment (nop33) [reply apps/mobile-wallet/src/components/NFTThumbnail.tsx:124]: I don't think we can, but I am open to ideas.

**Comment by nop33** [comment]
@mvaivre The things you mentioned already existed before I believe. Tanstack will probably be useful here.

File: apps/mobile-wallet/src/components/NFTThumbnail.tsx (hunk 2 above)
Comment (mvaivre) [reply apps/mobile-wallet/src/components/NFTThumbnail.tsx:124]: I don't have any, except for preventing from opening the full size image..

File: apps/mobile-wallet/src/components/NFTImage.tsx [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+  ) : (
+    <NFTImageStyled
+      style={{ width: size, height: size }}
+      transition={500}
+      source={{ uri: nft.image }}
+      allowDownscaling
+      contentFit="contain"
+      onError={() => setHasError(true)}
+      placeholder={{
+        blurhash: theme.name === 'dark' ? 'L00000fQfQfQfQfQfQfQfQfQfQfQ' : 'L1PGpx-;fQ-;_3fQfQfQfQfQfQfQ'
```
Comment (mvaivre) [inline apps/mobile-wallet/src/components/NFTImage.tsx:57]: wut?

Comment (mvaivre) [reply apps/mobile-wallet/src/components/NFTImage.tsx:57]: (that would be so cool to be able to use this :))

Comment (nop33) [reply apps/mobile-wallet/src/components/NFTImage.tsx:57]: yes, I love the idea of blurhash. Many of the NFT indexers provide it. Unfortunately, we can't use one :(

**Review by mvaivre — APPROVED** [review]
🎨

**Pull request merged**